packages/kbn-utility-types/src/dot_test.ts

```diff
@@ -0,0 +1,97 @@
+/*
```

Comment:
Should we name this file as .test instead of _test? Just thinking about the convention defined in the IDM RFC

Reply:
@afharo I intentionally named it _test instead of .test because it doesn't contain any unit tests, no need for the test runner to actually execute it. Does that make sense?

Reply:
Can't we use `expectType` for those tests, as it's done for the other tests of the package (see the tests in `packages/kbn-utility-types/src/tsd_tests/test_d`)